src/store/modules/keymap.js

```diff
@@ -238,6 +237,31 @@ const mutations = {
     }
     state.selectedIndex = index;
   },
+  /**
+   * Use this after switching out the keycodes for a new layout to keep the keymap
+   * in sync. It has it's own copy of the labels, which need to be updated for
+   * correct ISO support.
+   * @param {} state
+   */
+  updateKeycodeNames(state) {
```

Comment:
So this is the core of the new behavior. When requested, force the keymap to update all the keycode names from the latest lookup keycode table. This is what the UI uses to determine what to display. The keycodes don't change between locales. Only the name/label/displayname which is all derived for name. The only special handling here is for container keys, where it's the contained key that needs an update, not the container.